client/components/data/media-list-data/index.jsx

```diff
@@ -74,7 +79,11 @@ module.exports = React.createClass( {
 	},
 
 	updateStateData: function() {
-		this.setState( getStateData( this.props.siteId ) );
+		const state = getStateData( this.props.siteId );
+		this.setState( state );
+		if ( ! this.props.requestingMediaStorage && this.props.siteId ) {
```

Comment:
Could we add an extra condition here to trigger the request only when the media length changes?
eg `this.state.media.length !== state.media.length`